plugins/filetrees/yazi.nix

```diff
@@ -0,0 +1,161 @@
+          open_file_in_tab = "<c-t>";
+          grep_in_directory = "<c-s>";
+          replace_in_directory = "<c-g>";
+          cycle_open_buffers = "<tab>";
+          copy_relative_path_to_selected_files = "<c-y>";
+          send_to_quickfix_list = "<c-q>";
+        }
+        ''
+          Customize the keymaps that are active when yazi is open and focused.
+
+          Also:
+            - use e.g. `open_file_in_tab = false` to disable a keymap
+            - you can customize only some of the keymaps if you want
+            - Set to `false` to disable all default keymaps.
+        '';
+
+    set_keymappings_function = defaultNullOpts.mkLuaFn null ''
+      Completely override the keymappings for yazi. This function will be
+      called in the context of the yazi terminal buffer.
+    '';
+
+    hooks = {
+      yazi_opened = defaultNullOpts.mkLuaFn' {
+        pluginDefault.__raw = ''
+          function(preselected_path, yazi_buffer_id, config)
+          end
+        '';
+        description = ''
+          If you want to execute a custom action when yazi has been opened,
+          you can define it here.
+        '';
+      };
+
+      yazi_closed_successfully = defaultNullOpts.mkLuaFn' {
+        pluginDefault.__raw = ''
+          function(chosen_file, config, state)
+          end
+        '';
+        description = "When yazi was successfully closed";
+      };
+
+      yazi_opened_multiple_files = defaultNullOpts.mkLuaFn' {
+        pluginDefault.__raw = ''
+          function(chosen_files)
+            vim.cmd("args" .. table.concat(chosen_files, " "))
+          end
+        '';
+        description = ''
+          When yazi opened multiple files. The default is to send them to the
+          quickfix list, but if you want to change that, you can define it here
+        '';
+      };
+    };
+
+    highlight_groups = defaultNullOpts.mkAttributeSet { hovered_buffer = null; } ''
+      Add highlight groups to different yazi events.
+
+      NOTE: this only works if `use_ya_for_events_reading` is enabled, etc.
+    '';
+
+    floating_window_scaling_factor =
+      defaultNullOpts.mkNum 0.9
+        "The floating window scaling factor. 1 means 100%, 0.9 means 90%, etc.";
+
+    yazi_floating_window_winblend = defaultNullOpts.mkNullableWithRaw (types.ints.between 0
+      100
+    ) 0 "`0` for fully opaque and `100` for fully transparent. See :h winblend";
+
+    yazi_floating_window_border = defaultNullOpts.mkBorder "rounded" "yazi" ''
+      The type of border to use for the floating window.
+
+      Supports all available border types from `vim.api.keyset.win_config.border`.
+    '';
+  };
+
+  settingsExample = {
+    log_level = "debug";
+    open_for_directories = true;
+    enable_mouse_support = true;
+    floating_window_scaling_factor = 0.5;
+    yazi_floating_window_border = "single";
+    yazi_floating_window_winblend = 50;
+  };
+
+  extraPlugins = with pkgs.vimPlugins; [ plenary-nvim ];
```

Comment:
@GaetanLepage is this a dependency that should be upstreamed to nixpkgs? Or perhaps it's already handled there?

Reply:
I didn't add it when i added the plugin. Is that just adding a dependency in ovverrides.nix?
```nix
  yazi-nvim = super.yazi-nvim.overrideAttrs {
    dependencies = with self; [ plenary-nvim ];
  };
```

Reply:
> Is that just adding a dependency in overrides.nix?

Yes, I think so. Though @GaetanLepage is more familiar than I am with nixpkgs's vim/nvim contributions.

Reply:
Created a PR for the dependency upstream and added a TODO.